src/webgpu/api/validation/encoding/cmds/render/indirect_multi_draw.spec.ts

```diff
@@ -0,0 +1,409 @@
+      ? t.createBufferTracked({
+          size: 256,
+          usage: drawCountUsage,
+        })
+      : undefined;
+
+    const { encoder, validateFinish } = t.createEncoder('render pass');
+    encoder.setPipeline(t.createNoOpRenderPipeline());
+    if (indexed) {
+      encoder.setIndexBuffer(t.makeIndexBuffer(), 'uint32');
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (encoder as any).multiDrawIndexedIndirect(indirectBuffer, 0, 1, drawCountBuffer);
+    } else {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (encoder as any).multiDrawIndirect(indirectBuffer, 0, 1, drawCountBuffer);
+    }
+    const shouldSucceed =
+      (indirectUsage & GPUBufferUsage.INDIRECT) !== 0 &&
+      (!drawCountBuffer || drawCountUsage & GPUBufferUsage.INDIRECT) !== 0;
+    validateFinish(shouldSucceed);
+  });
+
+g.test('offsets_alignment')
+  .desc(
+    `
+Tests indirect and draw count offsets must be a multiple of 4.
+  `
+  )
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase('chromium-experimental-multi-draw-indirect' as GPUFeatureName);
+  })
+  .paramsSubcasesOnly(
+    kIndirectMultiDrawTestParams.combineWithParams([
+      // Valid
+      { indirectOffset: 0, drawCountOffset: 0 },
+      { indirectOffset: 4, drawCountOffset: 0 },
+      { indirectOffset: 0, drawCountOffset: 4 },
+      // Invalid
+      { indirectOffset: 2, drawCountOffset: 0 },
+      { indirectOffset: 6, drawCountOffset: 0 },
+      { indirectOffset: 0, drawCountOffset: 2 },
+      { indirectOffset: 0, drawCountOffset: 6 },
+    ] as const)
+  )
+  .fn(t => {
+    const { indexed, indirectOffset, useDrawCountBuffer, drawCountOffset } = t.params;
+
+    const indirectBuffer = t.createBufferTracked({
+      size: 256,
+      usage: GPUBufferUsage.INDIRECT,
+    });
+    const drawCountBuffer = useDrawCountBuffer
+      ? t.createBufferTracked({
+          size: 256,
+          usage: GPUBufferUsage.INDIRECT,
+        })
+      : undefined;
+
+    const { encoder, validateFinish } = t.createEncoder('render pass');
+    encoder.setPipeline(t.createNoOpRenderPipeline());
+    if (indexed) {
+      encoder.setIndexBuffer(t.makeIndexBuffer(), 'uint32');
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (encoder as any).multiDrawIndexedIndirect(
+        indirectBuffer,
+        indirectOffset,
+        1,
+        drawCountBuffer,
+        drawCountOffset
+      );
+    } else {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (encoder as any).multiDrawIndirect(
+        indirectBuffer,
+        indirectOffset,
+        1,
+        drawCountBuffer,
+        drawCountOffset
+      );
+    }
+
+    // We need to figure out if https://github.com/gpuweb/gpuweb/pull/2315/files#r1773031950 applies.
+    validateFinish(indirectOffset % 4 === 0 && (!useDrawCountBuffer || drawCountOffset % 4 === 0));
+  });
+
+g.test('indirect_buffer_OOB')
```

Comment:
nit: Realized this also validates alignment in addition to OOB, so more generally:
```suggestion
g.test('indirectBuffer_range')
```

also `indirectBuffer` because it refers to the argument called `indirectBuffer`.